juju/juju
# [JUJU-2568] Add cross model secret facade to support reading secret content from cross model relations

Add a new "cross model secrets" api facade implementation. This facade is a cross model controller facade with a method to fetch the content of a secret managed on another controller.

This PR does not use any authentication  - the client side api supports macaroons (and retry/cache) the same as other cross model apis. But there's no server side implementation yet - this will come next.

The secrets manager facade creates and calls out to this new facade if the source model of the secret uri is on another controller.

You can only share secrets from the offering side of a relation - this is enforced when the grant is created.

## QA steps

deploy juju-qa-dummy-source and juju-qa-dummy-sink charms in models on different controllers
offer dummy-source:sink and relate dummy-sink to the offer
in the offer charm, create a secret
grant access to the relation (-r 0)
from dummy-sink charm, get the secret value
remove the grant and get the value again to check permission denied
